src/plugins/data/public/ui/query_string_input/query_bar_top_row.tsx

```diff
@@ -348,7 +348,7 @@ function QueryBarTopRowUI(props: Props) {
       className={classes}
       responsive={!!props.showDatePicker}
       gutterSize="s"
-      justifyContent="flexEnd"
+      justifyContent="flexStart"
```

Comment:
I've made this change because the date picker looks out of place when it's right aligned and the query input is not displayed - mostly because the filter bar is left aligned below it. Is there any harm in doing this?

Reply:
I don't see any harm in doing this.

Reply:
I generally prefer the right alignment, and I feel keeping the date picker location consistent is a good thing. However, in embedded mode this may not be as important because the end user may not be a frequent Kibana user, and I think that it looks out of place when there is content on the lines above and below like this:
![flexEnd](https://user-images.githubusercontent.com/5101376/75495885-fd0acf80-5a02-11ea-97a0-beea66a2e896.png)

I can make this a prop so that it doesn't impact the other views, but it would be good to hear some more thoughts. I also have no strong opinion on this, but I made the change to find out if it's preferred. I'd be happy to change it back.

Reply:
I agree with you @ajwild. We are aware that this is not ideal and are continuing a search for a better solution. The better solution though involves lots of folks across different teams coming to a conclusion so it's taking a while to nail down. Consistency is the main factor here, and we should try to keep the global time filters in the same place across all apps (even when no query bar exists).

In embed mode 🤷‍♀ , it's probably okay to shift.

Reply:
@ajwild I like the prop solution in that case

Reply:
I think it's okay to do a shift, as long as it's not complicating the API \ code.
As the design of this will soon change anyway, I think we could go with an 80:20 solution, allowing having the datepicker in embedded mode, even if it's not perfectly positioned ATM.